From 2139e8f89cd2561f6f1125fa0ce25c9fa84519a7 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Mon, 15 Jan 2024 17:58:58 +0200
Subject: [PATCH] [SW-170792] gaudi3: pif: disable vpd capability

By default Vital Product Data (VPD) capability is enabled in PCIe
configuration space. This makes the host to send VPD requests which
by default are translated to unsupported transactions to the device,
which are translated to DMARs to HOST or LBW transactions to blocked
address of the device (depending on BMU configuration).
This capability is disabled.

Change-Id: If90876b2ff1a0a3a14ca2c6adb03f64015245110
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index f36b7c4..4021dab 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -1012,6 +1012,10 @@
 	regmap_writel(0x04020031, D0_PCIE_DBI_SIG_REG(mmPCIE_DBI_GEN3_EQ_CONTROL_OFF));
 	regmap_rmwritel(D0_PCIE_DBI_SIG_REG(mmPCIE_DBI_DEVICE_CAPABILITIES2_REG),
 					ENABLE_PWRBRK_CTRL_VAL, EMR_PWR_M);
+
+	/* remove Vital Product Data (VPD) reporting capability */
+	regmap_rmwritel(D0_PCIE_DBI_SIG_REG(mmPCIE_DBI_PCI_MSIX_CAP_ID_NEXT_CTRL_REG), 0,
+			PCIE_DBI_PCI_MSIX_CAP_ID_NEXT_CTRL_REG_PCI_MSIX_CAP_NEXT_OFFSET_M);
 }
 
 __weak void preboot_update_bar_sizes(void) {}
